pandas/sparse/frame.py

```diff
@@ -187,9 +213,47 @@ def _init_matrix(self, data, index, columns, dtype=None):
         if len(index) != N:
             raise ValueError('Index length mismatch: %d vs. %d' %
                              (len(index), N))
+        return index, columns
 
-        data = dict([(idx, data[:, i]) for i, idx in enumerate(columns)])
-        return self._init_dict(data, index, columns, dtype)
+    def to_coo(self):
+        """
+        Convert the frame to its SciPy sparse COO matrix representation.
+
+        .. versionadded:: 0.20.0
+
+        Returns
+        -------
+        coo_matrix : scipy.sparse.spmatrix
+            If the caller is heterogeneous and contains booleans or objects,
+            the result will be of dtype=object. See Notes.
+
+        Notes
+        -----
+        The dtype will be the lowest-common-denominator type (implicit
```

Comment:
As efficient as `ndarray.astype()` and `np.concatenate()` are. Iterated on the number of columns as that was the easiest for me to understand. Doesn't make an unnecessary copy since `copy=False` was added just now. Thanks.

What do you mean by referring main docs?